core/transaction.go

```diff
@@ -224,18 +279,69 @@ func (d *DeclareTransaction) Hash(network utils.Network) (*felt.Felt, error) {
 	return nil, errors.New("invalid transaction version")
 }
 
-type L1HandlerTransaction struct {
-	// todo
+func l1HandlerTransactionHash(l *L1HandlerTransaction, network utils.Network) (*felt.Felt, error) {
+	// TODO: implement me
+	panic("implement me")
 }
 
-func (h *L1HandlerTransaction) Hash(network utils.Network) (*felt.Felt, error) {
-	panic("not implemented")
+func deployAccountTransactionHash(d *DeployAccountTransaction,
+	network utils.Network,
+) (*felt.Felt, error) {
+	// TODO: implement me
+	panic("implement me")
 }
 
-type DeployAccountTransaction struct {
-	// todo
+const commitmentTrieHeight uint = 64
+
+// TransactionCommitment is the root of a height 64 binary Merkle Patricia tree of the
+// transaction hashes and signatures in a block.
+func TransactionCommitment(transactions []Transaction) (*felt.Felt, error) {
+	var transactionCommitment *felt.Felt
+	return transactionCommitment, trie.RunOnTempTrie(commitmentTrieHeight, func(trie *trie.Trie) error {
+		for i, transaction := range transactions {
+			signatureHash := crypto.PedersenArray()
+			if _, ok := transaction.(*InvokeTransaction); ok {
```

Comment:
if should be removed

Reply:
Not really. Although there are other transactions with Signatures, we only use that of `InvokeTransaction` to calculate commitments

Reply:
my bad